docs/tutorial/durham_reframe_tutorial.md

```diff
@@ -0,0 +1,409 @@
+def extract_scale_perf(self):
+    return sn.extractsingle(r'Scale:\s+(\S+)\s+.*', self.stdout, 1, float)
+
+@performance_function('MB/s', perf_key='Add')
+def extract_add_perf(self):
+    return sn.extractsingle(r'Add:\s+(\S+)\s+.*', self.stdout, 1, float)
+
+@performance_function('MB/s', perf_key='Triad')
+def extract_triad_perf(self):
+    return sn.extractsingle(r'Triad:\s+(\S+)\s+.*', self.stdout, 1, float)
+```
+
+----
+
+### Perflogs
+
+- Perflogs are output in `perflogs/<system>/<partition>`
+- By default a lot of information is printed. This can be customized in the configuration file. More on this later.
+- By default not much information about build step, has to be linked back to build environment
+- See `.reframe/reports/` or use `--report-file`
+
+----
+
+## Reference values
+
+ReFrame can automate checking that the results fall within an expected range. You can set a different reference value for each `perf_key` in the performance function. For example, set the test to fail if it falls outside of +-25% of the values obtained with the previous array size.
+
+```python
+reference = {
+    'cosma': {
+        'Copy':  (40000, -0.25, 0.25, 'MB/s'),
+        'Scale': (20000, -0.25, 0.25, 'MB/s'),
+        'Add':   (20000, -0.25, 0.25, 'MB/s'),
+        'Triad': (20000, -0.25, 0.25, 'MB/s')
+    }
+}
+```
+
+> The performance reference tuple consists of the reference value, the lower and upper thresholds expressed as fractional numbers relative to the reference value, and the unit of measurement. If any of the thresholds is not relevant, None may be used instead. Also, the units in this reference variable are entirely optional, since they were already provided through the @performance_function decorator.
+
+
+----
+
+### Parametrized tests
+
+You can pass a list to the `parameter()` built-in function in the class body to create a parametrized test. You cannot access the individual parameter value within the class body, so any reference to them should be placed in the appropriate function, for example `__init__()`
+
+For parametrisation you can add for example
+```python
+    arraysize = parameter([5,15,25])
+    self.build_system.cppflags = [f'-DSTREAM_ARRAY_SIZE=$((2 ** {self.arraysize}))']
+```
+
+You can have multiple parameters. ReFrame will run all parameter combinations by default.
+
+---
+
+## [Build systems](https://reframe-hpc.readthedocs.io/en/v4.5.2/tutorial_advanced.html#more-on-building-tests)
+- [Build systems Reference](https://reframe-hpc.readthedocs.io/en/v4.5.2/regression_test_api.html#build-systems)
+
+----
+
+### [Make](https://reframe-hpc.readthedocs.io/en/v4.5.2/tutorial_advanced.html#more-on-building-tests) 
+
+- Tutorial in `tutorials/advanced/makefiles/maketest.py`. 
+
+> you must set the executable attribute of the test, because ReFrame cannot know what is the actual executable to be run. We then set the build system to Make and set the preprocessor flags as we would do with the SingleSource build system.
```

Comment:
This paragraph makes no sense on its own. Either remove or explain.
Also, it's formatted as a comment.

Reply:
I included the whole paragraph in the quote, does it make more sense?